rust-keybroker/keybroker-server/src/verifier.rs

```diff
@@ -1,17 +1,58 @@
 // Copyright 2024 Contributors to the Veraison project.
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::error::{Result, VerificationErrorKind};
+use crate::error::{Error, Result, VerificationErrorKind};
 use crate::policy;
 use ear::{Algorithm, Ear};
 use veraison_apiclient::*;
 
-pub fn verify_with_veraison_instance(
+/// The trait that must be implemented to emit diagnostics for specific flavours of EAR.
+pub trait DiagnosticEmitter {
```

Comment:
Again just being picky, but the Rust convention appears to be to name traits after active verb-phrases rather than nouns. It is a departure from other languages. So I think it would be more idiomatic to call this something like `EmitDiagnostic`

Reply:
Addressed as well in the next patchset !